### PR TITLE
feat: add collapsibles for right panel contents

### PR DESCRIPTION
**Pull request by harish-sethuraman**
Fixes https://github.com/eslint/eslint.org/issues/310

### CODE REVIEW

**Comment by shanpriyan** [comment]
I think the chevron is pointing to the wrong direction and the color should be white in dark mode
(Your PR)
image

It should point upwards, if opened
(archived [eslint/playground](https://deploy-preview-92--play-eslint.netlify.app/) )
image

And also configurations are collapsed by default in [archived playground](https://deploy-preview-92--play-eslint.netlify.app/)

**Comment by harish-sethuraman** [comment]
> And also configurations are collapsed by default in [archived playground](https://deploy-preview-92--play-eslint.netlify.app/)

This Im unsure if we want to collapse by default... I think it would be better to have it open by default?

**Comment by shanpriyan** [comment]
> > And also configurations are collapsed by default in [archived playground](https://deploy-preview-92--play-eslint.netlify.app/)
> 
> This Im unsure if we want to collapse by default... I think it would be better to have it open by default?

collapsing by default brings the theme toggle button to viewport, other than that opened by default is fine I think.

May be we can collapse least used options 🤔?

**Comment by harish-sethuraman** [comment]
Yea we could collapse versioning and config I guess. Lets see what others have to say cc: @eslint/eslint-docs

**Comment by kecrily** [comment]
Let's collapse version control, by default we should always use the latest

Additionally, because folding will cause the slider to appear to squeeze the layout and appear to flicker. Need to add a style

```css
.playground__config-and-footer {
    scrollbar-gutter: stable;
}
```

| Before | After |
| - | - |
| 动画 | 
动画 |

**Comment by nzakas** [comment]
The share URL should also be collapsed by default. In the original design, only the rules section was expanded by default.

File: src/playground/components/ShareURL.js [hunk 1]
```diff
@@ -2,11 +2,15 @@ import React, { useState } from "react";
 
 export default function ShareURL({ url }) {
     const [isDataCopied, setIsDataCopied] = useState(false);
+    const [ShowShareURL, setShowShareURL] = useState(false);
```
Comment (shanpriyan) [inline src/playground/components/ShareURL.js:5]:
```suggestion
    const [showShareURL, setShowShareURL] = useState(false);
```

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

File: src/playground/components/ShareURL.js [hunk 2]
```diff
@@ -2,11 +2,15 @@ import React, { useState } from "react";
...
 
     return (
         <div>
-            <h2 data-config-section-title>Share URL</h2>
-            <div data-config-section>
+            <button className="playground-toggle c-btn c-btn--ghost" onClick={() => setShowShareURL(currentValue => !currentValue)}>
+                <h2 data-config-section-title>Share URL</h2>
+                <svg height="20" width="20" viewBox="0 0 20 20" aria-hidden="true" focusable="false" fill="currentColor" style={{ transform: ShowShareURL ? "rotate(180deg)" : null }}><path d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"></path></svg>
+            </button>
+            {ShowShareURL && <div data-config-section>
```
Comment (harish-sethuraman) [inline src/playground/components/ShareURL.js:13]:
```suggestion
            {showShareURL && <div data-config-section>
```

File: src/playground/components/ShareURL.js [hunk 3]
```diff
@@ -2,11 +2,15 @@ import React, { useState } from "react";
...
     const [isDataCopied, setIsDataCopied] = useState(false);
+    const [ShowShareURL, setShowShareURL] = useState(false);
 
     return (
         <div>
-            <h2 data-config-section-title>Share URL</h2>
-            <div data-config-section>
+            <button className="playground-toggle c-btn c-btn--ghost" onClick={() => setShowShareURL(currentValue => !currentValue)}>
+                <h2 data-config-section-title>Share URL</h2>
+                <svg height="20" width="20" viewBox="0 0 20 20" aria-hidden="true" focusable="false" fill="currentColor" style={{ transform: ShowShareURL ? "rotate(180deg)" : null }}><path d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"></path></svg>
```
Comment (harish-sethuraman) [inline src/playground/components/ShareURL.js:11]:
```suggestion
                <svg height="20" width="20" viewBox="0 0 20 20" aria-hidden="true" focusable="false" fill="currentColor" style={{ transform: showShareURL ? "rotate(180deg)" : null }}><path d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"></path></svg>
```

**Review by nzakas — APPROVED** [review]

**Pull request merged**